07.SubpassBaking/main.cpp

```diff
@@ -420,7 +423,12 @@ class SubpassBaking : public ApplicationBase
                         video::IGPUBuffer::SCreationParams scratchParams = {};
                         scratchParams.canUpdateSubRange = true;
                         scratchParams.usage = core::bitflag(video::IGPUBuffer::EUF_TRANSFER_DST_BIT) | video::IGPUBuffer::EUF_STORAGE_BUFFER_BIT;
-                        scratch = { 0ull,logicalDevice->createDeviceLocalGPUBufferOnDedMem(scratchParams,ppHandler->getMaxScratchSize()) };
+                        scratchParams.size = ppHandler->getMaxScratchSize();
+                        auto scratchBuff = logicalDevice->createBuffer(scratchParams);
+                        auto memReqs = scratchBuff->getMemoryReqs();
+                        memReqs.size = scratchParams.size;
```

Comment:
Wrong,

You already have the size from the `buffer->getMemReqs()` you do not need to fill it again, what you need is 
`memReqs.memoryTypeBits &= physdev->getDeviceLocalMemoryTypeBits` -> BECAUSE `logicalDevice->createDeviceLocalGPUBufferOnDedMem` was used previously